ui/src/auth/Purgatory.js

```diff
@@ -1,70 +1,164 @@
-import React, {PropTypes} from 'react'
+import React, {Component, PropTypes} from 'react'
 import {connect} from 'react-redux'
+import {bindActionCreators} from 'redux'
+import {withRouter} from 'react-router'
+
+import {meChangeOrganizationAsync} from 'shared/actions/auth'
 
 import {MEMBER_ROLE} from 'src/auth/Authorized'
 
-const memberCopy = (
-  <p>This role does not grant you sufficient permissions to view Chronograf.</p>
-)
-const viewerCopy = (
-  <p>
-    This organization does not have any configured sources<br />and your role
-    does not have permission to configure a source.
-  </p>
-)
+const getRoleNameByOrgID = (id, roles) => {
+  const role = roles.find(r => r.organization === id)
+  return (role && role.name) || 'ghost'
+}
+
+class Purgatory extends Component {
+  handleClickLogin = organization => async e => {
+    e.preventDefault()
+    const {router, links, meChangeOrganization} = this.props
+
+    await meChangeOrganization(links.me, {organization: organization.id})
+    router.push('')
+  }
```

Comment:
This component is still stateless.  So, lets keep it as an SFC.  You can move `handleClickLogin` out of the component and pass in router, links, and meChangeOrganization as arguments.